core/src/main/java/com/linecorp/armeria/client/ArrayBasedEventLoopState.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.netty.channel.EventLoop;
+
+final class ArrayBasedEventLoopState extends AbstractEventLoopState {
+
+    private final List<AbstractEventLoopEntry> entries = new ArrayList<>();
+    private final int maxNumEventLoops;
+
+    private EventLoopAcquisitionIndex acquisitionIndex;
```

Comment:
```suggestion
    @Nullable
    private EventLoopAcquisitionIndex acquisitionIndex;
```
We need to add the annotation if the field is nullable so that IDEs do not complain about the usage.